### PR TITLE
bpo-35794: Catch PermissionError in test_no_such_executable

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue35794

### CODE REVIEW

File: Lib/test/test_posix.py [hunk 1]
```diff
@@ -1522,7 +1522,7 @@ def test_no_such_executable(self):
             pid = self.spawn_func(no_such_executable,
                                   [no_such_executable],
                                   os.environ)
-        except FileNotFoundError as exc:
+        except (FileNotFoundError, PermissionError) as exc:
```
Comment (vstinner) [inline Lib/test/test_posix.py:1527]: Maybe add a comment mentioning bpo-35794 to explain why PermissionError is expected.

**Review by vstinner — APPROVED** [review]
LGTM, maybe just add proposed comment.

File: Lib/test/test_posix.py [hunk 2]
```diff
@@ -1522,7 +1522,9 @@ def test_no_such_executable(self):
             pid = self.spawn_func(no_such_executable,
                                   [no_such_executable],
                                   os.environ)
-        except FileNotFoundError as exc:
+        # bpo-35794: PermissionError can be raised if there are
+        # directories in the $PATH that are not accesible.
```
Comment (vstinner) [inline Lib/test/test_posix.py]: typo: accessible?

Comment (pablogsal) [reply Lib/test/test_posix.py]: Wops! Fixed in last commit

**Pull request merged**